src/megaclient.cpp

```diff
@@ -8128,6 +8223,22 @@ void MegaClient::disabletransferresumption(const char *loggedoutid)
 
 void MegaClient::fetchnodes(bool nocache)
 {
+    if (fetchingnodes)
+    {
+        return;
+    }
+
+    WAIT_CLASS::bumpds();
+    fnstats.init();
+    if (sid.size() >= SIDLEN)
+    {
+        fnstats.type = FetchNodesStats::TYPE_ACCOUNT;
+    }
+    else if (publichandle != UNDEF)
```

Comment:
I think that the best would be that all those checks could rely on MegaClient::loggedin(), creating a new value in the enumeration it returns. That would require a review of all usages of that function and changes not related to this pull request at all, so I think that it's better to change it in another one.